apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/environment/PortalMetaDomainConsts.java

```diff
@@ -0,0 +1,182 @@
+package com.ctrip.framework.apollo.portal.environment;
+
+import com.ctrip.framework.apollo.core.utils.ApolloThreadFactory;
+import com.ctrip.framework.apollo.core.utils.NetUtil;
+import com.ctrip.framework.apollo.tracer.Tracer;
+import com.ctrip.framework.apollo.tracer.spi.Transaction;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Only use in apollo-portal
+ * Provider an available meta server url.
+ * If there is no available meta server url for the given environment,
+ * the default meta server url will be used(http://apollo.meta).
+ * @see com.ctrip.framework.apollo.core.MetaDomainConsts
+ * @author wxq
+ */
+public class PortalMetaDomainConsts {
```

Comment:
Please add unit test for PortalMetaDomainConsts

Reply:
In `apollo-core`, there is a test for `MetaDomain`, i.e `com.ctrip.framework.apollo.core.MetaDomainTest`. It seems that implements `com.ctrip.framework.apollo.BaseIntegrationTest`, and start a internal  server to test.

If i want to write the unit test of `com.ctrip.framework.apollo.portal.environment.PortalMetaDomainConsts`, should i use the same way?

A little Challenge to do this. (hard)

Reply:
I think we need to do it and it looks like you have done that part

Reply:
The unit test is done.

Reply:
Most code is copied from `apollo-core`